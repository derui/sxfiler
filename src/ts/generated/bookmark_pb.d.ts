// package: 
// file: bookmark.proto

import * as jspb from "google-protobuf";

export class Bookmark extends jspb.Message {
  getName(): string;
  setName(value: string): void;

  getPath(): string;
  setPath(value: string): void;

  serializeBinary(): Uint8Array;
  toObject(includeInstance?: boolean): Bookmark.AsObject;
  static toObject(includeInstance: boolean, msg: Bookmark): Bookmark.AsObject;
  static extensions: {[key: number]: jspb.ExtensionFieldInfo<jspb.Message>};
  static extensionsBinary: {[key: number]: jspb.ExtensionFieldBinaryInfo<jspb.Message>};
  static serializeBinaryToWriter(message: Bookmark, writer: jspb.BinaryWriter): void;
  static deserializeBinary(bytes: Uint8Array): Bookmark;
  static deserializeBinaryFromReader(message: Bookmark, reader: jspb.BinaryReader): Bookmark;
}

export namespace Bookmark {
  export type AsObject = {
    name: string,
    path: string,
  }
}

export class BookmarkList extends jspb.Message {
  clearItemsList(): void;
  getItemsList(): Array<Bookmark>;
  setItemsList(value: Array<Bookmark>): void;
  addItems(value?: Bookmark, index?: number): Bookmark;

  serializeBinary(): Uint8Array;
  toObject(includeInstance?: boolean): BookmarkList.AsObject;
  static toObject(includeInstance: boolean, msg: BookmarkList): BookmarkList.AsObject;
  static extensions: {[key: number]: jspb.ExtensionFieldInfo<jspb.Message>};
  static extensionsBinary: {[key: number]: jspb.ExtensionFieldBinaryInfo<jspb.Message>};
  static serializeBinaryToWriter(message: BookmarkList, writer: jspb.BinaryWriter): void;
  static deserializeBinary(bytes: Uint8Array): BookmarkList;
  static deserializeBinaryFromReader(message: BookmarkList, reader: jspb.BinaryReader): BookmarkList;
}

export namespace BookmarkList {
  export type AsObject = {
    itemsList: Array<Bookmark.AsObject>,
  }
}

export class ListAllRequest extends jspb.Message {
  serializeBinary(): Uint8Array;
  toObject(includeInstance?: boolean): ListAllRequest.AsObject;
  static toObject(includeInstance: boolean, msg: ListAllRequest): ListAllRequest.AsObject;
  static extensions: {[key: number]: jspb.ExtensionFieldInfo<jspb.Message>};
  static extensionsBinary: {[key: number]: jspb.ExtensionFieldBinaryInfo<jspb.Message>};
  static serializeBinaryToWriter(message: ListAllRequest, writer: jspb.BinaryWriter): void;
  static deserializeBinary(bytes: Uint8Array): ListAllRequest;
  static deserializeBinaryFromReader(message: ListAllRequest, reader: jspb.BinaryReader): ListAllRequest;
}

export namespace ListAllRequest {
  export type AsObject = {
  }
}

export class ListAllResponse extends jspb.Message {
  hasBookmarks(): boolean;
  clearBookmarks(): void;
  getBookmarks(): BookmarkList | undefined;
  setBookmarks(value?: BookmarkList): void;

  serializeBinary(): Uint8Array;
  toObject(includeInstance?: boolean): ListAllResponse.AsObject;
  static toObject(includeInstance: boolean, msg: ListAllResponse): ListAllResponse.AsObject;
  static extensions: {[key: number]: jspb.ExtensionFieldInfo<jspb.Message>};
  static extensionsBinary: {[key: number]: jspb.ExtensionFieldBinaryInfo<jspb.Message>};
  static serializeBinaryToWriter(message: ListAllResponse, writer: jspb.BinaryWriter): void;
  static deserializeBinary(bytes: Uint8Array): ListAllResponse;
  static deserializeBinaryFromReader(message: ListAllResponse, reader: jspb.BinaryReader): ListAllResponse;
}

export namespace ListAllResponse {
  export type AsObject = {
    bookmarks?: BookmarkList.AsObject,
  }
}

export class RegisterRequest extends jspb.Message {
  getPath(): string;
  setPath(value: string): void;

  serializeBinary(): Uint8Array;
  toObject(includeInstance?: boolean): RegisterRequest.AsObject;
  static toObject(includeInstance: boolean, msg: RegisterRequest): RegisterRequest.AsObject;
  static extensions: {[key: number]: jspb.ExtensionFieldInfo<jspb.Message>};
  static extensionsBinary: {[key: number]: jspb.ExtensionFieldBinaryInfo<jspb.Message>};
  static serializeBinaryToWriter(message: RegisterRequest, writer: jspb.BinaryWriter): void;
  static deserializeBinary(bytes: Uint8Array): RegisterRequest;
  static deserializeBinaryFromReader(message: RegisterRequest, reader: jspb.BinaryReader): RegisterRequest;
}

export namespace RegisterRequest {
  export type AsObject = {
    path: string,
  }
}

export class RegisterResponse extends jspb.Message {
  hasBookmark(): boolean;
  clearBookmark(): void;
  getBookmark(): Bookmark | undefined;
  setBookmark(value?: Bookmark): void;

  serializeBinary(): Uint8Array;
  toObject(includeInstance?: boolean): RegisterResponse.AsObject;
  static toObject(includeInstance: boolean, msg: RegisterResponse): RegisterResponse.AsObject;
  static extensions: {[key: number]: jspb.ExtensionFieldInfo<jspb.Message>};
  static extensionsBinary: {[key: number]: jspb.ExtensionFieldBinaryInfo<jspb.Message>};
  static serializeBinaryToWriter(message: RegisterResponse, writer: jspb.BinaryWriter): void;
  static deserializeBinary(bytes: Uint8Array): RegisterResponse;
  static deserializeBinaryFromReader(message: RegisterResponse, reader: jspb.BinaryReader): RegisterResponse;
}

export namespace RegisterResponse {
  export type AsObject = {
    bookmark?: Bookmark.AsObject,
  }
}

export class DeleteRequest extends jspb.Message {
  getId(): string;
  setId(value: string): void;

  serializeBinary(): Uint8Array;
  toObject(includeInstance?: boolean): DeleteRequest.AsObject;
  static toObject(includeInstance: boolean, msg: DeleteRequest): DeleteRequest.AsObject;
  static extensions: {[key: number]: jspb.ExtensionFieldInfo<jspb.Message>};
  static extensionsBinary: {[key: number]: jspb.ExtensionFieldBinaryInfo<jspb.Message>};
  static serializeBinaryToWriter(message: DeleteRequest, writer: jspb.BinaryWriter): void;
  static deserializeBinary(bytes: Uint8Array): DeleteRequest;
  static deserializeBinaryFromReader(message: DeleteRequest, reader: jspb.BinaryReader): DeleteRequest;
}

export namespace DeleteRequest {
  export type AsObject = {
    id: string,
  }
}

export class DeleteResponse extends jspb.Message {
  hasDeletedBookmarks(): boolean;
  clearDeletedBookmarks(): void;
  getDeletedBookmarks(): Bookmark | undefined;
  setDeletedBookmarks(value?: Bookmark): void;

  serializeBinary(): Uint8Array;
  toObject(includeInstance?: boolean): DeleteResponse.AsObject;
  static toObject(includeInstance: boolean, msg: DeleteResponse): DeleteResponse.AsObject;
  static extensions: {[key: number]: jspb.ExtensionFieldInfo<jspb.Message>};
  static extensionsBinary: {[key: number]: jspb.ExtensionFieldBinaryInfo<jspb.Message>};
  static serializeBinaryToWriter(message: DeleteResponse, writer: jspb.BinaryWriter): void;
  static deserializeBinary(bytes: Uint8Array): DeleteResponse;
  static deserializeBinaryFromReader(message: DeleteResponse, reader: jspb.BinaryReader): DeleteResponse;
}

export namespace DeleteResponse {
  export type AsObject = {
    deletedBookmarks?: Bookmark.AsObject,
  }
}

