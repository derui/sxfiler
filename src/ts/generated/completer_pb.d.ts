// package: completer
// file: completer.proto

import * as jspb from "google-protobuf";

export class Item extends jspb.Message {
  getId(): string;
  setId(value: string): void;

  getValue(): string;
  setValue(value: string): void;

  serializeBinary(): Uint8Array;
  toObject(includeInstance?: boolean): Item.AsObject;
  static toObject(includeInstance: boolean, msg: Item): Item.AsObject;
  static extensions: {[key: number]: jspb.ExtensionFieldInfo<jspb.Message>};
  static extensionsBinary: {[key: number]: jspb.ExtensionFieldBinaryInfo<jspb.Message>};
  static serializeBinaryToWriter(message: Item, writer: jspb.BinaryWriter): void;
  static deserializeBinary(bytes: Uint8Array): Item;
  static deserializeBinaryFromReader(message: Item, reader: jspb.BinaryReader): Item;
}

export namespace Item {
  export type AsObject = {
    id: string,
    value: string,
  }
}

export class Candidate extends jspb.Message {
  getStart(): number;
  setStart(value: number): void;

  getLength(): number;
  setLength(value: number): void;

  hasValue(): boolean;
  clearValue(): void;
  getValue(): Item | undefined;
  setValue(value?: Item): void;

  serializeBinary(): Uint8Array;
  toObject(includeInstance?: boolean): Candidate.AsObject;
  static toObject(includeInstance: boolean, msg: Candidate): Candidate.AsObject;
  static extensions: {[key: number]: jspb.ExtensionFieldInfo<jspb.Message>};
  static extensionsBinary: {[key: number]: jspb.ExtensionFieldBinaryInfo<jspb.Message>};
  static serializeBinaryToWriter(message: Candidate, writer: jspb.BinaryWriter): void;
  static deserializeBinary(bytes: Uint8Array): Candidate;
  static deserializeBinaryFromReader(message: Candidate, reader: jspb.BinaryReader): Candidate;
}

export namespace Candidate {
  export type AsObject = {
    start: number,
    length: number,
    value?: Item.AsObject,
  }
}

export class InitializeRequest extends jspb.Message {
  clearSourceList(): void;
  getSourceList(): Array<Item>;
  setSourceList(value: Array<Item>): void;
  addSource(value?: Item, index?: number): Item;

  serializeBinary(): Uint8Array;
  toObject(includeInstance?: boolean): InitializeRequest.AsObject;
  static toObject(includeInstance: boolean, msg: InitializeRequest): InitializeRequest.AsObject;
  static extensions: {[key: number]: jspb.ExtensionFieldInfo<jspb.Message>};
  static extensionsBinary: {[key: number]: jspb.ExtensionFieldBinaryInfo<jspb.Message>};
  static serializeBinaryToWriter(message: InitializeRequest, writer: jspb.BinaryWriter): void;
  static deserializeBinary(bytes: Uint8Array): InitializeRequest;
  static deserializeBinaryFromReader(message: InitializeRequest, reader: jspb.BinaryReader): InitializeRequest;
}

export namespace InitializeRequest {
  export type AsObject = {
    sourceList: Array<Item.AsObject>,
  }
}

export class InitializeResponse extends jspb.Message {
  serializeBinary(): Uint8Array;
  toObject(includeInstance?: boolean): InitializeResponse.AsObject;
  static toObject(includeInstance: boolean, msg: InitializeResponse): InitializeResponse.AsObject;
  static extensions: {[key: number]: jspb.ExtensionFieldInfo<jspb.Message>};
  static extensionsBinary: {[key: number]: jspb.ExtensionFieldBinaryInfo<jspb.Message>};
  static serializeBinaryToWriter(message: InitializeResponse, writer: jspb.BinaryWriter): void;
  static deserializeBinary(bytes: Uint8Array): InitializeResponse;
  static deserializeBinaryFromReader(message: InitializeResponse, reader: jspb.BinaryReader): InitializeResponse;
}

export namespace InitializeResponse {
  export type AsObject = {
  }
}

export class CompleteRequest extends jspb.Message {
  getInput(): string;
  setInput(value: string): void;

  serializeBinary(): Uint8Array;
  toObject(includeInstance?: boolean): CompleteRequest.AsObject;
  static toObject(includeInstance: boolean, msg: CompleteRequest): CompleteRequest.AsObject;
  static extensions: {[key: number]: jspb.ExtensionFieldInfo<jspb.Message>};
  static extensionsBinary: {[key: number]: jspb.ExtensionFieldBinaryInfo<jspb.Message>};
  static serializeBinaryToWriter(message: CompleteRequest, writer: jspb.BinaryWriter): void;
  static deserializeBinary(bytes: Uint8Array): CompleteRequest;
  static deserializeBinaryFromReader(message: CompleteRequest, reader: jspb.BinaryReader): CompleteRequest;
}

export namespace CompleteRequest {
  export type AsObject = {
    input: string,
  }
}

export class CompleteResponse extends jspb.Message {
  serializeBinary(): Uint8Array;
  toObject(includeInstance?: boolean): CompleteResponse.AsObject;
  static toObject(includeInstance: boolean, msg: CompleteResponse): CompleteResponse.AsObject;
  static extensions: {[key: number]: jspb.ExtensionFieldInfo<jspb.Message>};
  static extensionsBinary: {[key: number]: jspb.ExtensionFieldBinaryInfo<jspb.Message>};
  static serializeBinaryToWriter(message: CompleteResponse, writer: jspb.BinaryWriter): void;
  static deserializeBinary(bytes: Uint8Array): CompleteResponse;
  static deserializeBinaryFromReader(message: CompleteResponse, reader: jspb.BinaryReader): CompleteResponse;
}

export namespace CompleteResponse {
  export type AsObject = {
  }
}

export class CompletionResultNotificationRequest extends jspb.Message {
  clearCandidatesList(): void;
  getCandidatesList(): Array<Candidate>;
  setCandidatesList(value: Array<Candidate>): void;
  addCandidates(value?: Candidate, index?: number): Candidate;

  serializeBinary(): Uint8Array;
  toObject(includeInstance?: boolean): CompletionResultNotificationRequest.AsObject;
  static toObject(includeInstance: boolean, msg: CompletionResultNotificationRequest): CompletionResultNotificationRequest.AsObject;
  static extensions: {[key: number]: jspb.ExtensionFieldInfo<jspb.Message>};
  static extensionsBinary: {[key: number]: jspb.ExtensionFieldBinaryInfo<jspb.Message>};
  static serializeBinaryToWriter(message: CompletionResultNotificationRequest, writer: jspb.BinaryWriter): void;
  static deserializeBinary(bytes: Uint8Array): CompletionResultNotificationRequest;
  static deserializeBinaryFromReader(message: CompletionResultNotificationRequest, reader: jspb.BinaryReader): CompletionResultNotificationRequest;
}

export namespace CompletionResultNotificationRequest {
  export type AsObject = {
    candidatesList: Array<Candidate.AsObject>,
  }
}

export class CompletionResultNotificationResponse extends jspb.Message {
  serializeBinary(): Uint8Array;
  toObject(includeInstance?: boolean): CompletionResultNotificationResponse.AsObject;
  static toObject(includeInstance: boolean, msg: CompletionResultNotificationResponse): CompletionResultNotificationResponse.AsObject;
  static extensions: {[key: number]: jspb.ExtensionFieldInfo<jspb.Message>};
  static extensionsBinary: {[key: number]: jspb.ExtensionFieldBinaryInfo<jspb.Message>};
  static serializeBinaryToWriter(message: CompletionResultNotificationResponse, writer: jspb.BinaryWriter): void;
  static deserializeBinary(bytes: Uint8Array): CompletionResultNotificationResponse;
  static deserializeBinaryFromReader(message: CompletionResultNotificationResponse, reader: jspb.BinaryReader): CompletionResultNotificationResponse;
}

export namespace CompletionResultNotificationResponse {
  export type AsObject = {
  }
}

