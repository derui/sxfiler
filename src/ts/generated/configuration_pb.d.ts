// package:
// file: configuration.proto

import * as jspb from "google-protobuf";
import * as types_pb from "./types_pb";

export class Configuration extends jspb.Message {
  getDefaultsortorder(): types_pb.SortTypeMap[keyof types_pb.SortTypeMap];
  setDefaultsortorder(value: types_pb.SortTypeMap[keyof types_pb.SortTypeMap]): void;

  serializeBinary(): Uint8Array;
  toObject(includeInstance?: boolean): Configuration.AsObject;
  static toObject(includeInstance: boolean, msg: Configuration): Configuration.AsObject;
  static extensions: { [key: number]: jspb.ExtensionFieldInfo<jspb.Message> };
  static extensionsBinary: { [key: number]: jspb.ExtensionFieldBinaryInfo<jspb.Message> };
  static serializeBinaryToWriter(message: Configuration, writer: jspb.BinaryWriter): void;
  static deserializeBinary(bytes: Uint8Array): Configuration;
  static deserializeBinaryFromReader(message: Configuration, reader: jspb.BinaryReader): Configuration;
}

export namespace Configuration {
  export type AsObject = {
    defaultsortorder: types_pb.SortTypeMap[keyof types_pb.SortTypeMap];
  };
}

export class GetRequest extends jspb.Message {
  serializeBinary(): Uint8Array;
  toObject(includeInstance?: boolean): GetRequest.AsObject;
  static toObject(includeInstance: boolean, msg: GetRequest): GetRequest.AsObject;
  static extensions: { [key: number]: jspb.ExtensionFieldInfo<jspb.Message> };
  static extensionsBinary: { [key: number]: jspb.ExtensionFieldBinaryInfo<jspb.Message> };
  static serializeBinaryToWriter(message: GetRequest, writer: jspb.BinaryWriter): void;
  static deserializeBinary(bytes: Uint8Array): GetRequest;
  static deserializeBinaryFromReader(message: GetRequest, reader: jspb.BinaryReader): GetRequest;
}

export namespace GetRequest {
  export type AsObject = {};
}

export class GetResponse extends jspb.Message {
  hasConfiguration(): boolean;
  clearConfiguration(): void;
  getConfiguration(): Configuration | undefined;
  setConfiguration(value?: Configuration): void;

  serializeBinary(): Uint8Array;
  toObject(includeInstance?: boolean): GetResponse.AsObject;
  static toObject(includeInstance: boolean, msg: GetResponse): GetResponse.AsObject;
  static extensions: { [key: number]: jspb.ExtensionFieldInfo<jspb.Message> };
  static extensionsBinary: { [key: number]: jspb.ExtensionFieldBinaryInfo<jspb.Message> };
  static serializeBinaryToWriter(message: GetResponse, writer: jspb.BinaryWriter): void;
  static deserializeBinary(bytes: Uint8Array): GetResponse;
  static deserializeBinaryFromReader(message: GetResponse, reader: jspb.BinaryReader): GetResponse;
}

export namespace GetResponse {
  export type AsObject = {
    configuration?: Configuration.AsObject;
  };
}

export class StoreRequest extends jspb.Message {
  hasConfiguration(): boolean;
  clearConfiguration(): void;
  getConfiguration(): Configuration | undefined;
  setConfiguration(value?: Configuration): void;

  serializeBinary(): Uint8Array;
  toObject(includeInstance?: boolean): StoreRequest.AsObject;
  static toObject(includeInstance: boolean, msg: StoreRequest): StoreRequest.AsObject;
  static extensions: { [key: number]: jspb.ExtensionFieldInfo<jspb.Message> };
  static extensionsBinary: { [key: number]: jspb.ExtensionFieldBinaryInfo<jspb.Message> };
  static serializeBinaryToWriter(message: StoreRequest, writer: jspb.BinaryWriter): void;
  static deserializeBinary(bytes: Uint8Array): StoreRequest;
  static deserializeBinaryFromReader(message: StoreRequest, reader: jspb.BinaryReader): StoreRequest;
}

export namespace StoreRequest {
  export type AsObject = {
    configuration?: Configuration.AsObject;
  };
}

export class StoreResponse extends jspb.Message {
  serializeBinary(): Uint8Array;
  toObject(includeInstance?: boolean): StoreResponse.AsObject;
  static toObject(includeInstance: boolean, msg: StoreResponse): StoreResponse.AsObject;
  static extensions: { [key: number]: jspb.ExtensionFieldInfo<jspb.Message> };
  static extensionsBinary: { [key: number]: jspb.ExtensionFieldBinaryInfo<jspb.Message> };
  static serializeBinaryToWriter(message: StoreResponse, writer: jspb.BinaryWriter): void;
  static deserializeBinary(bytes: Uint8Array): StoreResponse;
  static deserializeBinaryFromReader(message: StoreResponse, reader: jspb.BinaryReader): StoreResponse;
}

export namespace StoreResponse {
  export type AsObject = {};
}
