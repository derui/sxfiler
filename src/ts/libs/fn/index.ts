// define functional library in TypeScript

// pipe multiple function at once
export const pipe = function pipe<T extends any[], R>(fn1: (...args: T) => R, ...fns: Array<(arg: R) => R>) {
  const piped = fns.reduce(
    (current, next) => (value) => next(current(value)),
    (value) => value
  );

  return (...args: T) => piped(fn1(...args));
};

// compose two function
export const compose = function compose<V, R, N>(fn1: (arg: V) => R, fn2: (arg: R) => N) {
  return (v: V) => fn2(fn1(v));
};

/**
 * port Kotlin's `also` extend function.
 *
 * WARN: This function be able to modify the object in function. Avoid to use this funciton
 *       for shared function.
 */
export const also = function also<T>(it: T, f: (it: T) => void): T {
  f(it);
  return it;
};
