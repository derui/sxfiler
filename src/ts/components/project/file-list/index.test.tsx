import { h } from "preact";
import { render } from "preact-render-to-string";

import { Component as T } from "./index";

import { also } from "@/libs/fn";
import { FileItem, FileStat, Mode, Capability } from "@/generated/filer_pb";
import { Bookmark } from "@/generated/bookmark_pb";

function makeNode(name: string) {
  const item = new FileItem();
  item.setName(name);
  item.setMarked(true);
  item.setStat(
    also(new FileStat(), (v) => {
      v.setIsDirectory(false);
      v.setIsSymlink(false);
      v.setIsFile(true);
      v.setMode(
        also(new Mode(), (v) => {
          v.setOwner(
            also(new Capability(), (v) => {
              v.setReadable(true);
              v.setWritable(true);
            })
          );
        })
      );
      v.setAtime("0");
      v.setCtime("0");
      v.setMtime("0");
      v.setSize("10");
    })
  );
  return item;
}

const bookmarks = [
  also(new Bookmark(), (v) => {
    v.setName("bookmark");
    v.setPath("fullpath");
  }).toObject(),
];

describe("Project", () => {
  describe("Node List", () => {
    it("should not print before resized", () => {
      const nodes = [makeNode("file.txt").toObject()];
      const tree = render(<T items={nodes} location="loc" cursor={0} focused={false} bookmarks={bookmarks} />);

      expect(tree).toMatchSnapshot();
    });

    it("should select a node locating same the index of cursor when focused", () => {
      const nodes = [makeNode("file.txt").toObject()];
      const tree = render(<T items={nodes} location="loc" cursor={0} focused={true} bookmarks={bookmarks} />);

      expect(tree).toMatchSnapshot();
    });

    it("should show dummy content when nodes is empty", () => {
      const tree = render(<T items={[]} location="loc" cursor={0} focused={true} bookmarks={bookmarks} />);

      expect(tree).toMatchSnapshot();
    });
  });
});
