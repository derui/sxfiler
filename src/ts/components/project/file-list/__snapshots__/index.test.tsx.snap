// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project Node List should not print before resized 1`] = `"<div class=\\"file-list__root\\" data-testid=\\"fileList-root\\"><header class=\\"file-list__header\\" data-testid=\\"fileList-header\\">loc</header><div style=\\"overflow: visible;\\" class=\\"file-list__autosizer\\"></div></div>"`;

exports[`Project Node List should select a node locating same the index of cursor when focused 1`] = `"<div class=\\"file-list__root\\" data-testid=\\"fileList-root\\"><header class=\\"file-list__header\\" data-testid=\\"fileList-header\\" data-focused>loc</header><div style=\\"overflow: visible;\\" class=\\"file-list__autosizer\\"></div></div>"`;

exports[`Project Node List should show dummy content when nodes is empty 1`] = `"<div class=\\"file-list__root\\" data-testid=\\"fileList-root\\"><header class=\\"file-list__header\\" data-testid=\\"fileList-header\\" data-focused>loc</header><div style=\\"overflow: visible;\\" class=\\"file-list__autosizer\\"></div></div>"`;
